src/slog_test.rs

```diff
@@ -191,25 +193,74 @@ pub fn check_expected_stats(logs: &[serde_json::Value], mut expected_stats: Vec<
 }
 
 // LCOV_EXCL_START Don't test derives
+/// ExpectedStatSnapshot helper.
+/// E.g.
+/// ExpectedStatSnapshot {
+///     name: "test_group_bucket_counter",
+///     description: "Test cumulative bucket counter with groups",
+///     stat_type: BucketCounter,
+///     values: vec![
+///         ExpectedStatSnapshotValue {
+///             group_values: vec!["one".to_string(), "two".to_string()],
+///             bucket_limit: Some(BucketLimit::Num(-8)),
+///             value: 0f64,
+///         },
+///         ExpectedStatSnapshotValue {
+///             group_values: vec!["one".to_string(), "two".to_string()],
+///             bucket_limit: Some(BucketLimit::Num(0)),
+///             value: 0f64,
+///         },
+///         ExpectedStatSnapshotValue {
+///             group_values: vec!["one".to_string(), "two".to_string()],
+///             bucket_limit: Some(BucketLimit::Unbounded),
+///             value: 3f64,
+///         },
+///         ExpectedStatSnapshotValue {
+///             group_values: vec!["three".to_string(), "four".to_string()],
+///             bucket_limit: Some(BucketLimit::Num(-8)),
+///             value: 4f64,
+///         },
+///         ExpectedStatSnapshotValue {
+///             group_values: vec!["three".to_string(), "four".to_string()],
+///             bucket_limit: Some(BucketLimit::Num(0)),
+///             value: 4f64,
+///         },
+///         ExpectedStatSnapshotValue {
+///             group_values: vec!["three".to_string(), "four".to_string()],
+///             bucket_limit: Some(BucketLimit::Unbounded),
+///             value: 4f64,
+///         },
+///     ],
+///     buckets: Some(Buckets::new(BucketMethod::CumulFreq, "bucket", &[-8, 0])),
+/// }
 #[derive(Debug)]
 pub struct ExpectedStatSnapshot {
+    /// Name of the stat
```

Comment:
Sigh.  Rust;s `mandatory comments for public struct fields` annoys me because you end up with pointless comments like this that tell you nothing useful and just take up vertical screen space.  But appreciate that fixing up warnings is good...

Reply:
It winds me up a bit too when this happens, but it does mean the autogenned rust docs don't having missing bits.